PayPalWebPayments/src/main/java/com/paypal/android/paypalwebpayments/PayPalWebLauncher.kt

```diff
@@ -0,0 +1,162 @@
+package com.paypal.android.paypalwebpayments
+
+import android.net.Uri
+import androidx.fragment.app.FragmentActivity
+import com.braintreepayments.api.BrowserSwitchClient
+import com.braintreepayments.api.BrowserSwitchOptions
+import com.braintreepayments.api.BrowserSwitchResult
+import com.braintreepayments.api.BrowserSwitchStatus
+import com.paypal.android.corepayments.APIClientError
+import com.paypal.android.corepayments.CoreConfig
+import com.paypal.android.corepayments.Environment
+import com.paypal.android.corepayments.PayPalSDKError
+import com.paypal.android.paypalwebpayments.PayPalWebCheckoutClient.Companion.DEEP_LINK_PARAM_APPROVAL_SESSION_ID
+import com.paypal.android.paypalwebpayments.PayPalWebCheckoutClient.Companion.DEEP_LINK_PARAM_APPROVAL_TOKEN_ID
+import com.paypal.android.paypalwebpayments.PayPalWebCheckoutClient.Companion.VAULT_HOST
+import com.paypal.android.paypalwebpayments.errors.PayPalWebCheckoutError
+import org.json.JSONObject
+
+internal class PayPalWebLauncher(
+    private val urlScheme: String,
+    private val coreConfig: CoreConfig,
+    private val browserSwitchClient: BrowserSwitchClient = BrowserSwitchClient(),
+) {
+    private val redirectUriPayPalCheckout = "$urlScheme://x-callback-url/paypal-sdk/paypal-checkout"
+
+    companion object {
+        const val METADATA_KEY_ORDER_ID = "order_id"
+        const val METADATA_KEY_SETUP_TOKEN_ID = "setup_token_id"
+    }
+
+    fun launchPayPalWebCheckout(
+        activity: FragmentActivity,
+        request: PayPalWebCheckoutRequest,
+    ): PayPalSDKError? {
+        var error: PayPalSDKError? = null
+        try {
+            val browserSwitchOptions = request.run {
+                configurePayPalBrowserSwitchOptions(orderId, coreConfig, fundingSource)
+            }
+            browserSwitchClient.start(activity, browserSwitchOptions)
+        } catch (e: PayPalSDKError) {
+            error = APIClientError.clientIDNotFoundError(e.code, e.correlationId)
+        }
+        return error
+    }
+
+    fun launchPayPalWebVault(activity: FragmentActivity, vaultRequest: PayPalWebVaultRequest) {
+        val browserSwitchOptions = vaultRequest.run {
+            configurePayPalVaultApproveSwitchOptions(setupTokenId, approveVaultHref)
+        }
+        browserSwitchClient.start(activity, browserSwitchOptions)
```

Comment:
This method throws a `BrowserSwitchException`, should we catch that here and return a PayPalSDKError? Or update the method to throw that error as well? (Same comment for `launchPayPalWebCheckout`)

Reply:
Yeah good call. I noticed that it seems like we never did catch this error on PPCP. It's tough because all exceptions are unchecked on the Kotlin side, so we don't get a compiler warning in this case.